react/PlaceDetails.tsx

```diff
@@ -26,8 +28,10 @@ const PlaceDetails: React.FC<Props> = ({
   display = 'extended',
   hiddenFields = [],
   onEdit,
+  customAddress,
 }) => {
-  const { address, rules } = useAddressContext()
+  const { address: contextAddress, rules } = useAddressContext()
```

Comment:
[@question](https://conventionalcomments.org): what is this `rules`?

Reply:
`rules` are a set of address rules and settings specific to a country ([AddressRules typing](https://github.com/vtex-apps/address-context/blob/61cbaf1ed0a7cdbba942f8cd36a46647b12cb50c/react/types.tsx#L71-L73)). For example, you can check the brazilian rules [here](https://github.com/vtex-apps/countries-data/blob/main/packages/bra/vtex.country-data-settings/configuration.json). This file includes things used for input validation (like CPF, phone, etc) and which components we should render on screen. To better visualize this, the property [display](https://github.com/vtex-apps/countries-data/blob/main/packages/bra/vtex.country-data-settings/configuration.json#L167) on extended mode describes the format that we want to show, like in the pictures bellow. For UK users, it makes no sense to display number, as it is already handled by address line 1, so we hide this from the user.

All this information comes from the `useAddressContext`

BR | UK
-|-
![image](https://user-images.githubusercontent.com/26108090/111547805-1686d280-8750-11eb-8b5c-90051dfa3b31.png) | ![image](https://user-images.githubusercontent.com/26108090/111547846-2dc5c000-8750-11eb-85e6-2de3e460006a.png)

Obs: first picture is outdated with the current setting (there is no `receiverName` property) and UK rules haven't been created yet.

Reply:
Thanks for this great explanation!

Reply:
Agree, awesome explanation 🙂

Maybe a good place to write down some words about what these `rules` mean could be the README of this project: https://github.com/vtex-apps/address-context

Super happy that you are taking good care of this project Wes! 😸